libres/tests/res_util/test_memory.cpp

```diff
@@ -0,0 +1,107 @@
+#include "catch2/catch.hpp"
+#include <fmt/format.h>
+
+#include <sstream>
+
+#include <ert/logging.hpp>
+#include <ert/res_util/memory.hpp>
+
+static int callno = 0;
+
+/*
+ * "Mock" the internal function which reads memory-info from
+ * the /proc-filesystem.
+ *
+ * Note that although the real memory-info is only available
+ * on Linux, this "mocking" makes the test run also on MacOs
+ */
+namespace ert {
+namespace utils {
+std::shared_ptr<std::istream> get_file(const char *filename) {
+    std::string s = R"(
+Name:	test
+Umask:	0022
+State:	R (running)
+Tgid:	72786
+Ngid:	0
+Pid:	72786
+PPid:	1
+TracerPid:	0
+Uid:	0	0	0	0
+Gid:	0	0	0	0
+FDSize:	256
+Groups:
+NStgid:	72786
+NSpid:	72786
+NSpgid:	72786
+NSsid:	1
+VmPeak:	  160560 kB
+VmSize:	  146680 kB
+VmLck:	       0 kB
+VmPin:	       0 kB
+VmHWM:	    7892 kB
+VmRSS:	    7892 kB
+RssAnon:	    4680 kB
+RssFile:	    3212 kB
+RssShmem:	       0 kB
+VmData:	  141976 kB
+VmStk:	     132 kB
+VmExe:	    1628 kB
+VmLib:	       4 kB
+VmPTE:	      68 kB
+VmSwap:	       0 kB
+nonvoluntary_ctxt_switches:	4)";
+
+    /* This method is called four times while the test runs.
+   * The last two calls are upon exiting the scope,
+   * hence simulate changes
+   * */
+    if (callno > 1) {
+        std::string fnd1("160560 kB"), fnd2("146680 kB");
+        std::string rep1("260560 kB"), rep2("46680 kB");
+
+        // Pretend VmPeak increased
+        s.replace(s.find(fnd1), fnd1.size(), rep1);
+        // Pretend VmSize decreased
+        s.replace(s.find(fnd2), fnd2.size(), rep2);
+    }
+    callno++;
+
+    return std::make_shared<std::istringstream>(s);
+}
+} // namespace utils
+} // namespace ert
+
+class MockLogger : public ert::ILogger {
+public:
+    virtual ~MockLogger() = default;
+    std::vector<std::string> calls;
+
+protected:
+    void log(ert::ILogger::Level level, fmt::string_view f,
+             fmt::format_args args) {
```

Comment:
Mark this function as overridden (ie, implementation of a `virtual` function):

```suggestion
    void log(ert::ILogger::Level level, fmt::string_view f,
             fmt::format_args args) override {
```